code/code/misc/riding.cc

```diff
@@ -823,8 +847,18 @@ int MountEgoCheck(TBeing *ch, TBeing *horse)
     check = horse->GetMaxLevel();
     check += dynamic_cast<TMonster *>(horse)->anger() / 30;
     check -= ch->GetMaxLevel();
+
     if (ch->doesKnowSkill(SKILL_RIDE))
       check -= ch->getSkillValue(SKILL_RIDE)/10;
+
+    // Bonus for proficiency in advanced riding disc.
+    if (ch->doesKnowSkill(SKILL_ADVANCED_RIDING))
+      check -= ch->getSkillValue(SKILL_ADVANCED_RIDING)/8;
+
+    // Bonus for proficiency in type-specific mount ability
+    for (const auto skill : {SKILL_RIDE_WINGED, SKILL_RIDE_DOMESTIC, SKILL_RIDE_NONDOMESTIC, SKILL_RIDE_EXOTIC })
+      check -= horse->mountSkillType() == skill && ch->doesKnowSkill(skill) ? ch->getSkillValue(skill)/6 : 0;
```

Comment:
Hey, truly sorry, but the previous version only checked as many items as necessary to find a match. Now we loop over them all - even if the mount is WINGED, we still try them all until EXOTIC.

We could add `{ check -= ... ; break}`.

Reply:
Reworked this (and learned a bit more about ternary operators).  I moved away from the ternary operator since I don't think I can include the break within one as it's not an expression.  Let me know if this new format looks acceptable.  Thanks!

Reply:
I think your ternary statement would've worked fine if you just put the break on the next line after it, instead. But the way you have it now actually reads nicer anyway, I think. Ternarys can be super useful but can also impact legibility a lot of times.